SQL-Solutions.txt

```diff
@@ -0,0 +1,2743 @@
+
+Query -
+
+SELECT 
+    salespeople.sname, orders.snum, SUM(orders.amt) AS total_amt
+FROM
+    orders
+        INNER JOIN
+    salespeople ON salespeople.snum = orders.snum
+GROUP BY orders.snum
+HAVING SUM(orders.amt) > (SELECT 
+        MAX(amt)
+    FROM
+        orders);
+
+Output -
+
++-------+------+-----------+
+| sname | snum | total_amt |
++-------+------+-----------+
+| Peel  | 1001 |  15382.07 |
++-------+------+-----------+
+
+-- 103) Give names and numbers of all salespersons that have more than one customer? 
+
+Query -
+
+SELECT 
+    snum, sname
+FROM
+    salespeople
+WHERE
+    snum IN (SELECT 
+            snum
+        FROM
+            customers
+        GROUP BY snum
+        HAVING COUNT(*) > 1);
+
+Output -
+
++------+--------+
+| snum | sname  |
++------+--------+
+| 1001 | Peel   |
+| 1002 | Serres |
++------+--------+
+
+-- 104) Select all salespeople by name and number who have customers in 
+-- their city whom they don’t service. 
+
+Query -
+
+SELECT 
+    sname, snum
+FROM
+    salespeople
+WHERE
+    EXISTS( SELECT 
+            *
+        FROM
+            customers
+        WHERE
+            salespeople.city = customers.city
+                AND salespeople.snum != customers.snum);
+
+Output -
+
++--------+------+
+| sname  | snum |
++--------+------+
+| Serres | 1002 |
+| Motika | 1004 |
+| Fran   | 1008 |
++--------+------+
+
+-- 105) Does the total amount in orders by customer in Rome and London, exceed the
+-- commission paid to salesperson in London, and New York by more than 5 times?
+
+Query -
+
+SELECT 
+    (SELECT 
+            SUM(orders.amt)
+        FROM
+            orders,
+            customers
+        WHERE
+            customers.cnum = orders.cnum
+                AND customers.city IN ('Rome' , 'London')) AS total_amt_in_orders_by_customers_in_rome_london,
+    (SELECT 
+            SUM(total_comm) * 5 AS five_x_comm
+        FROM
+            (SELECT 
+                (SUM(orders.amt) / 100) * salespeople.comm AS total_comm
+            FROM
+                orders, salespeople
+            WHERE
+                orders.snum = salespeople.snum
+                    AND salespeople.city IN ('London' , 'New York')
+            GROUP BY orders.snum) comm_of_salespersons_in_london_newyork) AS five_x_comm;
+
+Output - 
+
++-------------------------------------------------+--------------------+
+| total_amt_in_orders_by_customers_in_rome_london | five_x_comm        |
++-------------------------------------------------+--------------------+
+|                              18995.399999999998 | 11130.911999999998 |
++-------------------------------------------------+--------------------+
+
+-- ANSWER IS YES
+
+-- 106) Which are the date, order number, amt and city for each salesperson
+-- (by name) for themaximum order he has obtained? 
+
+Query -
+
+SELECT 
+    salespeople.sname,
+    salespeople.snum,
+    orders.odate,
+    orders.onum,
+    orders.amt,
+    salespeople.city
+FROM
```

Comment:
Every  Where Cross Join?Is Cross join is so efficent  that you have used every where?

Reply:
Changed with inner join